src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WPF/AppDotXamlDocument.cs

```diff
@@ -0,0 +1,944 @@
+            return string.Empty;
+        }
+
+        // Escape any double quotes
+
+        // Make as content of an element
+        string xml = "<a>" + value + "</a>";
+        StringReader stringReader = new(xml);
+        XmlReaderSettings settings = new() { ConformanceLevel = ConformanceLevel.Fragment };
+        XmlReader xmlReader = XmlReader.Create(stringReader, settings);
+        xmlReader.ReadToFollowing("a");
+
+        string content = xmlReader.ReadElementContentAsString();
+        return content;
+    }
+
+    /// <summary>
+    /// Finds the Application element that all application.xaml files must include as
+    /// the single root node.
+    /// </summary>
+    private static void MoveToApplicationRootElement(XmlTextReader reader)
+    {
+        // XAML files must have only one root element.  For app.xaml, it must be "Application"
+        if (reader.MoveToContent() == XmlNodeType.Element
+            && reader.Name == ApplicationElementName)
+        {
+            // Okay
+            return;
+        }
+
+        throw new XamlReadWriteException(string.Format(VSResources.WPFApp_Xaml_CouldntFindRootElement_1, ApplicationElementName));
+    }
+
+    /// <summary>
+    /// Creates an XmlTextReader for the text
+    /// in the buffer.
+    /// </summary>
+    private XmlTextReader CreateXmlTextReader()
+    {
+        System.Diagnostics.Debug.Assert(_debugBufferLockCount > 0, "Should be using BufferLock!");
+        StringReader stringReader = new(GetAllText());
+        // Required by Fxcop rule CA3054 - DoNotAllowDTDXmlTextReader
+        XmlTextReader xmlTextReader = new(stringReader) { DtdProcessing = DtdProcessing.Prohibit };
+        return xmlTextReader;
+    }
+
+    /// <summary>
+    /// Finds the value of the given property inside the xaml file.  If
+    /// the property is not set in the xaml, an empty string is returned.
+    /// </summary>
+    private string GetApplicationPropertyValue(string propertyName)
+    {
+        using BufferLock bufferLock = new(_vsTextLines, this);
+
+        XmlTextReader reader = CreateXmlTextReader();
+        XamlProperty? prop = FindApplicationPropertyInXaml(reader, propertyName);
+        if (prop is not null)
+        {
+            return prop.UnescapedValue;
+        }
+        else
+        {
+            return string.Empty;
+        }
+    }
+
+    /// <summary>
+    /// Find the closing angle bracket on a single line.
+    /// See comments on FindClosingAngleBracket.
+    /// </summary>
+    /// <returns>The index on the line where the closing angle bracket is found, or -1 if not found.</returns>
+    private static int FindClosingAngleBracketHelper(string line)
+    {
+        int index = 0;
+
+        const char singleQuote = '\'';
+        const char doubleQuote = '"';
+        while (index < line.Length)
+        {
+            // Find the next character of interest
+            index = line.IndexOfAny(new[] { singleQuote, doubleQuote, '/', '>' }, index);
```

Comment:
This array is allocated potentially many times per line of the file. Perhaps pull it out to a static field.